File: Frontend/app/src/main/java/com/example/view/api/ApiService.java
package com.example.view.api;

import com.example.view.model.LoginRequest;
import com.example.view.model.LoginResponse;
import com.example.view.model.ProtectedResponse;
import com.example.view.model.SignupRequest;
import com.example.view.model.SignupResponse;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;

public interface ApiService {

    @POST("accounts/signup/")
    Call<SignupResponse> signup(@Body SignupRequest signupRequest);

    @POST("accounts/login/")
    Call<LoginResponse> login(@Body LoginRequest loginRequest);

    @GET("protected/")
    Call<ProtectedResponse> getProtectedData();


}

File: Meditation/Meditation_Frontend/app/src/main/java/com/example/viewproject/data/remote/api/ApiService.kt
package com.example.viewproject.data.remote.api

import com.example.viewproject.data.remote.dto.*
import retrofit2.http.Body
import retrofit2.http.Headers
import retrofit2.http.POST

interface ApiService {

    @Headers("Content-Type: application/json")
    @POST("api/accounts/login/")
    suspend fun login(@Body request: LoginRequest): UserResponse

    @POST("api/accounts/register/")
    suspend fun register(@Body request: RegisterRequest): UserResponse
}
